wallet-new/src/Cardano/Wallet/API/Response.hs

```diff
@@ -117,3 +105,12 @@ paginate PaginationParams{..} rawResultSet =
                                }
         slice                  = take pp . drop ((cp - 1) * pp) . toList
     in (slice rawResultSet, metadata)
+
+
+-- | Creates a 'WalletResponse' with just a single record into it.
+single :: a -> WalletResponse a
```

Comment:
Also, we could probably abstract over whether the response is `single` or a list with a typeclass and a function `walletResponse` - that `single` is visually annoying to me.

Reply:
>  that single is visually annoying to me.

It's just a plain function: what's annoying about it? 😉  Maybe you would like it to be named differently?

Reply:
It just seems to be boilerplate since most of our "regular" responses are `single`, nothing else.

Reply:
> It just seems to be boilerplate since most of our "regular" responses are single, nothing else.

Valid point, although one may argue it's only 6 chars more to write every time, so not too bad 😉  It's not obvious to me how to do it in a simple yet transparent way (well, there are plenty of options, but you pay a price one way or the other). Ok, here is my proposal: will crack on with merging the PR as-it-is (after I fix the style guide nitpicks) and open a ticket to discuss with the team how we can get rid of the boilerplate in a concide manner (aka having our cake and eating it too :) )